### ISSUE TITLE
Save amp-anaytics cookies in the highestAvailableDomain

### ISSUE THREAD

**Issue by ifatgv4** [ampproject/amphtml#21377]
We would like to save the cookies created by amp-analytics in the second level domain.
I see that such a functionality was added in the PR:

https://github.com/ampproject/amphtml/pull/1459/files

using the option: highestAvailableDomain

But it is not available in the amp-analytics tag.

Can it be added?

**Comment by calebcordry** [comment]
Routing FR to @zhouyx @jeffjose 

cc/  @ampproject/wg-analytics

**Comment by jeffjose** [comment]
Hi @ifatgv4, would you mind providing a little more context on what your usecase is?

**Comment by ifatgv4** [comment]
@jeffjose Yes, of course.

We need to save a query param in a cookie. The page can be either an AMP or a non-AMP page.
In a non-AMP page we save the cookie using our own tag. This tag saves the cookie in the second level domain.
In an AMP page, the AMP tag saves the cookie in the (full) domain.
This inconsistency causes us to have two versions of the cookie, in two different domains. 
If we could instruct the tag to save the cookie in the second level domain (or in the highestAvailableDomain, which the setCookie function supports) we could avoid this situation.

In short, what we need is something like this:

```javascript
<amp-analytics>
    <script type="application/json">
        {
                                    :
            "cookies": {
                "enabled": true,
                "highestAvailableDomain": true,
                "my_param_cookie": {
                    "value": "QUERY_PARAM(myParam)"
                }
            }        
        }
    </script>
</amp-analytics>
```

**Comment by ifatgv4** [comment]
Or another option (better, but not a must for us), would be to be able to set this option per cookie:

```javascript
<amp-analytics>
    <script type="application/json">
        {
                                    :
            "cookies": {
                "enabled": true,
                "my_param_cookie": {
                    "highestAvailableDomain": true,
                    "value": "QUERY_PARAM(myParam)"
                }
            }        
        }
    </script>
</amp-analytics>
```

**Comment by ifatgv4** [comment]
Hi, @jeffjose
Are you planning to support this?
Thanks

**Comment by lannka** [comment]
@ifatgv4 FYI we're priortizing this feature in the coming sprint

**Comment by lannka** [comment]
@ifatgv4 FYI, the change is merged and you will be able to test on the next canary build (tomorrow)

**Comment by ifatgv4** [comment]
@lannka, thank you very much!
